### PR TITLE
Fix the bug that drawCall is still submitted when opacity is 0

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/cocos-engine/issues/11210

### Changelog

*Fix the bug that drawCall is still submitted when opacity is 0

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/renderer/batcher-2d.ts [hunk 1]
```diff
@@ -794,41 +794,44 @@ export class Batcher2D implements IBatcher {
...
             // HARD COUPLING
-            updateOpacity(render.renderData, opacity);
-            const buffer = render.renderData.getMeshBuffer();
-            if (buffer) {
-                buffer.setDirty();
+                updateOpacity(render.renderData, opacity);
+                const buffer = render.renderData.getMeshBuffer();
+                if (buffer) {
+                    buffer.setDirty();
+                }
```
Comment (pandamicro) [inline cocos/2d/renderer/batcher-2d.ts:815]: If opacity turn to zero in the current frame, but the render data will potentially contain un-updated data, make sure you have decoupled data update and rendering very carefully

Comment (LinYunMo) [reply cocos/2d/renderer/batcher-2d.ts:815]: If the opacity of the current frame becomes 0, then the content in renderData will not be rendered and submitted, and there is no point in updating the data in renderData at this point. The component's colorDirty will not be reset until its opacity becomes non-zero, and then the renderData update logic will be performed.

File: cocos/2d/renderer/batcher-2d.ts [hunk 2]
```diff
@@ -794,41 +794,44 @@ export class Batcher2D implements IBatcher {
         // TODO Set opacity to ui property's opacity before remove it
         // @ts-expect-error temporary force set, will be removed with ui property's opacity
         uiProps._opacity = opacity;
-        if (uiProps.colorDirty) {
+        if (opacity > 0) {
```
Comment (SantyWang) [inline cocos/2d/renderer/batcher-2d.ts]: Use approx instead.

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 3]
```diff
@@ -123,7 +123,9 @@ void Batcher2d::walk(Node* node, float parentOpacity) { // NOLINT(misc-no-recurs
             entity->setColorDirty(false);
             entity->setVBColorDirty(true);
         }
-        if (entity->isEnabled()) {
+        if (entity->getOpacity() <= 0 ) {
+            breakWalk = true;
```
Comment (SantyWang) [inline native/cocos/2d/renderer/Batcher2d.cpp:127]: Shoule skip handlePostRender?

**Review by SantyWang — APPROVED** [review]

**Pull request merged**